### PR TITLE
MF-146: Update Notes Navigation and Structure

### PR DESCRIPTION
**Pull request by denniskigen**
This PR modifies the notes widget so that it corresponds with the routing and navigation structure specified in https://issues.openmrs.org/projects/MF/issues/MF-146.

Screenshot 2020-03-24 at 18 51 28

Screenshot 2020-03-20 at 21 27 44

Screenshot 2020-03-22 at 18 35 56

### CODE REVIEW

File: src/widgets/notes/notes-detailed-summary.component.tsx [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
 
   useEffect(() => {
     if (!isLoadingPatient && patient) {
       const subscription = getEncounterObservableRESTAPI(patientUuid).subscribe(
         (notes: any) => {
           setPatientNotes(notes.results);
-          setTotalPages(Math.ceil(notes.results.length / resultsPerPage));
-          setCurrentPageResults(notes.results.slice(0, resultsPerPage));
+          // setTotalPages(Math.ceil(notes.results.length / resultsPerPage));
+          // setCurrentPageResults(notes.results.slice(0, resultsPerPage));
```
Comment (donaldkibet) [inline src/widgets/notes/notes-detailed-summary.component.tsx]: should this be deleted?

File: src/widgets/notes/notes.test.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { cleanup, render, wait } from "@testing-library/react";
+import { BrowserRouter } from "react-router-dom";
+import Notes from "./notes.component";
+
+describe("<NotesComponent />", () => {
+  let wrapper: any;
+
+  afterEach(cleanup);
+
+  xit("renders without dying", async () => {
```
Comment (donaldkibet) [inline src/widgets/notes/notes.test.tsx]: we should not be skipping this test

File: src/widgets/notes/notes.test.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    wrapper = render(
+      <BrowserRouter>
+        <Notes />
+      </BrowserRouter>
+    );
+    await wait(() => {
+      expect(wrapper).toBeTruthy();
+    });
```
Comment (donaldkibet) [inline src/widgets/notes/notes.test.tsx]: I think this assertion is useless. `render()` always returns a wrapper under every circumstance, right? Or is that not true? The main thing that this test is doing is just verifying that `render()` doesn't throw an error - we don't need this line at all I don't think

**Review by donaldkibet — APPROVED** [review]
Nice work on this, a few comments

**Comment by donaldkibet** [comment]
> This PR modifies the notes widget so that it corresponds with the routing and navigation structure specified in https://issues.openmrs.org/projects/MF/issues/MF-146.
> 
> Screenshot 2020-03-20 at 22 17 31
> 
> Screenshot 2020-03-20 at 21 27 44

According to the designs **Details** section has audit info we should be following these designs across all detailed summary components. [Figma Design](https://www.figma.com/file/YyVhWTNp89bNiOGadYrHBk/Patient-Chart---Master-(current-file)?node-id=5145%3A160341)

File: src/widgets/notes/notes.test.tsx (hunk 2 above)
Comment (denniskigen) [reply src/widgets/notes/notes.test.tsx]: I'll be removing this test for now. I figure it'll be easier to test once we get this new workflow going for all the widgets.

File: src/widgets/notes/notes.test.tsx (hunk 3 above)
Comment (denniskigen) [reply src/widgets/notes/notes.test.tsx]: Yeah, it doesn't do much. I'll be removing this test for now as indicated above.

**Comment by denniskigen** [comment]
> > This PR modifies the notes widget so that it corresponds with the routing and navigation structure specified in https://issues.openmrs.org/projects/MF/issues/MF-146.
> > Screenshot 2020-03-20 at 22 17 31
> > Screenshot 2020-03-20 at 21 27 44
> 
> According to the designs **Details** section has audit info we should be following these designs across all detailed summary components. [Figma Design](https://www.figma.com/file/YyVhWTNp89bNiOGadYrHBk/Patient-Chart---Master-(current-file)?node-id=5145%3A160341)

The API response for this request doesn't provide audit information.

File: src/widgets/notes/notes-detailed-summary.component.tsx (hunk 1 above)
Comment (denniskigen) [reply src/widgets/notes/notes-detailed-summary.component.tsx]: Yeah, removed.

File: src/widgets/notes/notes-overview.component.tsx [hunk 4]
```diff
@@ -21,124 +21,129 @@ export default function NotesOverview(props: NotesOverviewProps) {
...
-                    style={{ textAlign: "end" }}
-                  >
-                    <svg
-                      className="omrs-icon"
-                      fill="var(--omrs-color-ink-low-contrast)"
-                    >
-                      <use xlinkHref="#omrs-icon-chevron-right" />
-                    </svg>
+                  <td>
+                    <Link to={`${match.path}/${note.uuid}`}>
```
Comment (jonathandick) [inline src/widgets/notes/notes-overview.component.tsx]: This should be notesPath, not match.path

**Review by jonathandick — CHANGES_REQUESTED** [review]

File: src/widgets/notes/notes-overview.component.tsx (hunk 4 above)
Comment (denniskigen) [reply src/widgets/notes/notes-overview.component.tsx]: Fixed

**Review by jonathandick — APPROVED** [review]

**Pull request merged**